packages/plugins/Pins/src/store/index.ts

```diff
@@ -209,31 +209,49 @@ const storeModule: PolarModule<PinsState, PinsState> = {
       { rootGetters, dispatch },
       coordinates: Coordinate
     ): Promise<boolean> {
-      const { boundaryLayerId, toastAction } =
+      const { boundaryLayerId, toastAction, boundaryOnError } =
         rootGetters?.configuration?.pins || {}
 
+      let boundaryCheckResult
+
       if (
         !boundaryLayerId ||
-        (await passesBoundaryCheck(
+        // if a setup error occured, client will act as if no boundaryLayerId specified
+        (boundaryCheckResult = await passesBoundaryCheck(
           rootGetters.map,
           boundaryLayerId,
           coordinates
-        ))
+        )) === true ||
+        (typeof boundaryCheckResult === 'symbol' &&
+          boundaryOnError !== 'strict') /* defaults to 'permissive' */
       ) {
         return true
       }
 
+      const errorOccurred = typeof boundaryCheckResult === 'symbol'
+
       if (toastAction) {
-        const info = {
-          type: 'info',
-          text: 'plugins.pins.toast.notInBoundary',
-          timeout: 10000,
-        }
-        dispatch(toastAction, info, { root: true })
+        const toast = errorOccurred
+          ? {
+              type: 'error',
+              text: 'plugins.pins.toast.boundaryError',
+              timeout: 0,
```

Comment:
I think it's good practice to let users close errors manually. Some information is missable, but this may be relevant regarding the client's behaviour and should not be easily dismissed.

Reply:
We called ☎️ and decided that from now on errors should have to be dismissed. This will be reflected in an ADR.